router/router.go

```diff
@@ -3,17 +3,31 @@ package router
 import (
 	"github.com/gin-gonic/gin"
 	"github.com/murilogilfelpeto/ps-tag-onboarding-go/configuration"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/handler"
 )
 
-var (
-	logger *configuration.Logger
-)
+var logger = configuration.NewLogger("router")
+
+type Router interface {
+	InitServer()
+	initializeRoutes(router *gin.Engine)
+}
+
+type router struct {
+	handler handler.Handler
+}
```

Comment:
The router is already defined by your framework library; you just need to define your routes and initialize the server. Consider handling server initialization in the `main` itself, while placing the route definition in a separate function to reduce boilerplate code. Aim for simplicity.

Reply:
If I add everything to main file, it will be a huge file that handles everything. I think that is more simple that the main file just call everything it needs to start and other files do their job